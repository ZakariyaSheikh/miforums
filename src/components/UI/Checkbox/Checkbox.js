import React from 'react';
import PropTypes from 'prop-types';

const Checkbox = ({
    onChange,
    name,
    id,
    disabled = false,
    className = 'checkbox',
}) => {
    return (
        <span className={className}>
            <label>
                <input
                    type='checkbox'
                    onChange={onChange}
                    name={name}
                    disabled={disabled}
                    id={id}
                ></input>
                {name}
            </label>
        </span>
    );
};

Checkbox.propTypes = {
    onChange: PropTypes.func.isRequired,
    name: PropTypes.string,
};

export default Checkbox;
