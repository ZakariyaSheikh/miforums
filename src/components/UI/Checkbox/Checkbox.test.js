import React from 'react';
import { render, cleanup, fireEvent, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Checkbox from './Checkbox';

afterEach(cleanup);

describe('</Checkbox>', () => {
    const mockProps = {
        onChange: jest.fn(),
        name: 'Test',
        id: 'Test',
        disabled: false,
        className: 'checkboxTest',
    };

    test('should render as expected', () => {
        // const { container } = render(<Checkbox {...mockProps} />);
        // expect(container.firstChild).toMatchSnapshot();
        const wrapper = renderer.create(<Checkbox {...mockProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should call onClick', () => {
        render(<Checkbox {...mockProps} />);
        fireEvent.click(screen.getByText('Test'));
        expect(mockProps.onChange).toHaveBeenCalled();
    });

    test('should have className', () => {
        const { container } = render(<Checkbox {...mockProps} />);
        expect(container.firstChild).toHaveClass('checkboxTest');
    });

    test('should be disabled', () => {
        const { getByRole } = render(
            <Checkbox {...mockProps} disabled={true} />
        );
        expect(getByRole('checkbox')).toBeDisabled();
    });
});
