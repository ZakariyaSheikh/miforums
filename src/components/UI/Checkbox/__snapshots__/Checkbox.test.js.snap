// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`</Checkbox> should render as expected 1`] = `
<span
  className="checkboxTest"
>
  <label>
    <input
      disabled={false}
      id="Test"
      name="Test"
      onChange={[MockFunction]}
      type="checkbox"
    />
    Test
  </label>
</span>
`;
